src/TensorFlowNET.Core/Buffers/Buffer.cs

```diff
@@ -14,45 +14,44 @@
    limitations under the License.
 ******************************************************************************/
 
+using NumSharp.Backends.Unmanaged;
 using System;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
-using NumSharp.Backends.Unmanaged;
+using Tensorflow.Util;
 using static Tensorflow.c_api;
 
 namespace Tensorflow
 {
     /// <summary>
     ///     Represents a TF_Buffer that can be passed to Tensorflow.
     /// </summary>
-    public class Buffer : DisposableObject
+    public sealed class Buffer : IDisposable
     {
-        private unsafe TF_Buffer buffer
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get => *bufferptr;
-        }
+        public SafeBufferHandle Handle { get; }
 
-        private unsafe TF_Buffer* bufferptr
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get => (TF_Buffer*) _handle;
-        }
+        /// <remarks>
+        /// <para>The result of this method is only valid when the underlying <see cref="Handle"/> has not been
+        /// disposed. If the lifetime of the object is unclear, a lease may be used to prevent disposal while the object
+        /// is in use. See <see cref="SafeHandleExtensions.Lease(SafeHandle)"/>.</para>
+        /// </remarks>
+        private unsafe ref readonly TF_Buffer DangerousBuffer
+            => ref Unsafe.AsRef<TF_Buffer>(Handle.DangerousGetHandle().ToPointer());
 
         /// <summary>
         ///     The memory block representing this buffer.
         /// </summary>
-        /// <remarks>The deallocator is set to null.</remarks>
-        public UnmanagedMemoryBlock<byte> MemoryBlock
+        /// <remarks>
+        /// <para>The deallocator is set to null.</para>
+        ///
+        /// <inheritdoc cref="SafeHandleLease" path="/devdoc/usage"/>
```

Comment:
@Oceania2018 I need to know if you prefer this form, or the form above. This case uses the new `inheritdoc` feature in Quick Info for Visual Studio 2019 to automatically include the same language everywhere you reference this (which also means if you change the original text in the `<devdoc>` element of `SafeHandleLease`, it will update everywhere). The advantage is it avoids copying the same text to many locations. The downside is it relies on an IDE feature to show the information to users. I can use whichever approach you prefer.

Reply:
I like it `<devdoc>`, I don't even know that before, glad to learn new things.

Reply:
> I don't even know that before, glad to learn new things.

This pull request is the very first time I've done this. 😄

Reply:
➡️ Both cases use the `inheritdoc` form now.